engine/datamanager.go

```diff
@@ -1367,7 +1366,7 @@ func (dm *DataManager) SetTrend(tr *Trend) (err error) {
 		return utils.ErrNoDatabaseConn
 	}
 	if dm.dataDB.GetStorageType() != utils.MetaInternal {
-		if err = tr.compress(dm.ms); err != nil {
+		if tr, err = tr.compress(dm.ms); err != nil {
```

Comment:
Please create here a new variable instead of overwriting the old one, to be cleaner. Ie: trStored.